### DISCUSSION TITLE
Ability to acquire loss for next token generation

### DISCUSSION THREAD

**Discussion by ramkrishna2910** [microsoft/onnxruntime-genai#367] [New features / APIs]
The generate function from huggingface returns language modelling [loss ](https://huggingface.co/docs/transformers/v4.40.1/en/main_classes/output#transformers.modeling_outputs.CausalLMOutput.loss)for next token generation. This loss helps calculating perplexity  of the model.
Is there a known way to acquire loss from the generate function in onnxruntime_genai?

**Comment by yufenglee** [comment]
Will add an api to support this.

**Comment by ramkrishna2910** [comment]
That would be great! Thanks :)

**Comment by ramkrishna2910** [comment]
Thanks for making this a discussion. 
Having onnxruntime_genai's generate function mirror the huggingface generate function would greatly improve ease of use.
For example, options like `max_new_tokens` are very useful.

**Comment by anutkk** [comment]
It's been implemented, although I'm not sure of how exactly this should be used:
https://github.com/microsoft/onnxruntime-genai/discussions/522
